tools/webdriver/webdriver/transport.py

```diff
@@ -13,67 +14,48 @@
 missing = object()
 
 
-class ResponseHeaders(dict):
+class ResponseHeaders(Mapping):
     """Read-only dictionary-like API for accessing response headers.
 
-    This class always returns all headers with the same name (separated by
-    commas). It doesn't ensure header types (e.g. binary vs string).
+    This class handles header case-insensitivity, and always returns all
+    headers with the same name (separated by commas). It doesn't ensure header
+    types (e.g. binary vs string).
     """
     def __init__(self, items):
+        self.headers_dict = {}
         for key, value in items:
             key = key.lower()
```

Comment:
So we are not preserving the case of keys here, which is probably fine, but we should definitely document it.

Reply:
Done.